src/WebJobs.Extensions.DurableTask/AzureStorageAccountProvider.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using DurableTask.AzureStorage;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask.Auth;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask.Options;
+using Microsoft.Extensions.Configuration;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Auth;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal sealed class AzureStorageAccountProvider : IStorageAccountProvider
+    {
+        private readonly IConnectionInfoResolver connectionInfoResolver;
+        private readonly IStorageCredentialsFactory credentialFactory;
+
+        public AzureStorageAccountProvider(IConnectionInfoResolver connectionInfoResolver, IStorageCredentialsFactory credentialFactory = null)
+        {
+            this.connectionInfoResolver = connectionInfoResolver ?? throw new ArgumentNullException(nameof(credentialFactory));
+            this.credentialFactory = credentialFactory ?? NullCredentialsFactory.Instance;
+        }
+
+        public StorageAccountDetails GetStorageAccountDetails(string name)
+        {
+            AzureStorageAccountOptions account = this.ResolveAccountInfo(name);
+            if (!string.IsNullOrEmpty(account.ConnectionString))
+            {
+                return new StorageAccountDetails { ConnectionString = account.ConnectionString };
+            }
```

Comment:
I've refactored how this is done. If there `AzureWebJobsStorage` has a value, we assume it is a connection string and simply parse it via `CloudStorageAccount.Parse(string)`. Otherwise, we attempt to read the other fields from the config section (eg. `AccountName`, etc) and use them to populate the most relevant constructor.